### ISSUE TITLE
Inconsistent types in tendermint RPC

### ISSUE THREAD

**Issue by IvantheTricourne** [f-o-a-m/kepler#92]
Related to #75.

## Missing Fields

Type definitions with fields that are missing when uninitialized

* [Info](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Response.hs#L97-L99): `last_block_height` and `last_block_app_hash`

## Null Fields

Type definitions with fields that are set to `null` when uninitialized

* [EvidenceList](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-tendermint-client/src/Network/Tendermint/Client.hs#L296)
* [Data](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-tendermint-client/src/Network/Tendermint/Client.hs#L289)

## String Fields

Type definitions with fields typed `"string"` (i.e., wrappers over number types)

[Client.hs](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-tendermint-client/src/Network/Tendermint/Client.hs)
* [Vote](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-tendermint-client/src/Network/Tendermint/Client.hs#L313): `height`, `round` and `index`
* [ResultTx](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-tendermint-client/src/Network/Tendermint/Client.hs#L135): `height`
* [RequestBlock](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-tendermint-client/src/Network/Tendermint/Client.hs#L97): `height_ptr`
* [RequestABCIQuery](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-tendermint-client/src/Network/Tendermint/Client.hs#L66): `height`

[Response.hs](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Response.hs)
* [DeliverTx](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Response.hs#L390): `gas_wanted` and `gas_used`
* [CheckTx](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Response.hs#L321): `gas_wanted` and `gas_used`
* [Query](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Response.hs#L222): `height` and `index`
* [Info](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Response.hs#L90): `height` and `app_version`

[Request.hs](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Request.hs)
* [EndBlock](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Request.hs#L350): `height`
* [Query](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/Request.hs#L206): `height`

[FieldTypes.hs](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs)
* [Evidence](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L560): `height` and `total_voting_power`
* [Header](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L477): `height`, `num_txs`, and `total_txs`
* [Version](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L450): `block` and `app`
* [PartSetHeader](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L397): `total`
* [LastCommitInfo](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L367): `round`
* [Validator](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L313): `power`
* [ValidatorUpdate](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L286): `power`
* [EvidenceParams](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L181): `max_age`
* [BlockParams](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-types/src/Network/ABCI/Types/Messages/FieldTypes.hs#L154): `max_bytes` and `max_gas`

[Types.hs](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-sdk/src/Tendermint/SDK/Router/Types.hs)
* [QueryResult](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-sdk/src/Tendermint/SDK/Router/Types.hs#L41): `height` and `index`
* [QueryArgs](https://github.com/f-o-a-m/hs-abci/blob/client-tests/hs-abci-sdk/src/Tendermint/SDK/Router/Types.hs#L34): `block_height`

**Comment by martyall** [comment]
related to #109

**Comment by alexfmpe** [comment]
Edit: Ok, this was resolved by bumping kepler, so I think it can be ignored.

I'm running into 
```
*** Exception: ParsingException "parsing Word64 failed, expected Number, but encountered String"
```
when trying to use `RPC.abciInfo` even though manually curling works
```
{
  "jsonrpc": "2.0",
  "id": "",
  "result": {
    "response": {
      "data": "1",
      "version": "123",
      "app_version": "1",
      "last_block_height": "123"
    }
  }
}
```

I see this was erroring as of https://github.com/f-o-a-m/kepler/pull/84/commits/92b3b4c3af88a80cbafc5927947464dc03b32c3f. 

I'm currently depending on an older version than that, and was wondering if bumping Kepler would fix this.

It seems to work in current Travis? https://travis-ci.com/github/f-o-a-m/kepler/jobs/310896828
```
RPC Request
{
    "jsonrpc": "2.0",
    "params": [],
    "method": "abci_info",
    "id": 2149148217020253723
}
RPC Response
{
    "result": {
        "response": {
            "data": "{\"size\":0}",
            "app_version": "1",
            "version": "0.16.1"
        }
    },
    "jsonrpc": "2.0",
    "id": 2149148217020253696
}
    Can query /abci_info and parse the result
```

**Comment by martyall** [comment]
Yeah this kind of stuff was always a pain to get right, we only lazily discover these kinds of errors. Let me know if you're problem wasn't actually fixed by the bump
